src/test/ui/issues/issue-65383.rs

```diff
@@ -0,0 +1,7 @@
+fn main() {
```

Comment:
Could use a comment saying this used to ICE and why it did.
Also relate it to the code by mentioning e.g. `check_expr_break`.
Also can we place the test with the other break-related tests?